File: 2788_sep_words/sep.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

/**
 * C++ 做这个就没有 python 那么简单了
 * 我还错了两次...
 * runtime: 81.19%
 * memory:  63.62%
 */
class Solution {
public:
    vector<string> splitWordsBySeparator(vector<string>& words, char separator) {
        std::vector<string> result;
        for (auto& word: words) {
            if (word.back() != separator)
                word.push_back(separator);
            int wlen = word.size();
            for (int ep = 0, sp = 0; ep < wlen; ep++) {
                char v = word[ep];
                if (v == separator) {
                    if (ep - sp && word[sp] != separator)
                        result.push_back(word.substr(sp, ep - sp));
                    sp = ep + 1;
                    while (sp < wlen && word[sp] == separator)
                        sp ++;
                    ep = sp;
                }
            }
        }
        return result;
    }
};

int main(int argc, char** argv) {
    Solution sol;

    return 0;
}

File: 2788_sep_words/sep.py
from typing import List

"""
    不愧是简单题，随便秒杀... 唉 希望中等和困难也可以多一点秒杀题
"""
class Solution:
    def splitWordsBySeparator(self, words: List[str], separator: str) -> List[str]:
        result = []
        for word in words:
            seps = word.split(separator)
            seps = list(filter(lambda x: len(x), seps))
            result.extend(seps)
        return result